paddle/fluid/platform/cuda_device_function.h

```diff
@@ -46,9 +46,16 @@ template <>
 __forceinline__ __device__ float16 CudaShuffleDownSync(unsigned mask,
                                                        float16 val, int delta,
                                                        int width) {
-  half tmp = static_cast<half>(val);
-  __shfl_down(tmp, static_cast<unsigned>(delta), width);
-  return float16(tmp);
```

Comment:
The CudaShuffleDownSync must be an atomic operator, otherwise will get the wrong result.